tests/test_objective_funs.py

```diff
@@ -296,33 +297,37 @@ def build(self, eq, use_jit=True, verbose=1):
             super().build(eq, use_jit, verbose)
 
         def compute(self, R_lmn):
-            return self.target * self.weight + self.y * R_lmn**3
+            return 200 + self.target * self.weight - self.y * R_lmn**3
```

Comment:
Can you explain the changes in this test? not sure I follow all of them

Reply:
This test checks to make sure that `target` and `weight` get updated after re-JIT-compiling. In the old API, the `_Objective.compute` function was JIT-compiled, but in the new API it is the `_Objective.compute_scaled` function that gets JIT-compiled. So even if the `DummyObjective.compute()` function was not modified, the tests would still fail because of the new functionality. Since I had to update the "correct" test values anyways, I decided to modify the compute function to make the scale of the numbers more reasonable. 

I also added new checks to the test related to how `DummyObjective.compute()` can give inconsistent behavior from `DummyObjective.compute_scaled()` before re-JIT-compiling.